sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -395,7 +395,10 @@ class CacheManager extends Logging with AdaptiveSparkPlanHelper {
    */
   private def getOrCloneSessionWithConfigsOff(session: SparkSession): SparkSession = {
     if (session.conf.get(SQLConf.CAN_CHANGE_CACHED_PLAN_OUTPUT_PARTITIONING)) {
-      session
+      // Bucketed scan only has one time overhead but can have multi-times benefits in cache,
+      // so we always do bucketed scan in a cached plan.
+      SparkSession.getOrCloneSessionWithConfigsOff(
+        session, SQLConf.AUTO_BUCKETED_SCAN_ENABLED :: Nil)
```

Comment:
Please update the method description too according to this code change.

https://github.com/apache/spark/blob/a21e19b6e7ac4c4c77b39d93a2da2cbe1c88c4c8/sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala#L394

Reply:
addressed